compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1434,13 +1709,53 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             debug!(?collected_lifetimes);
             debug!(?new_remapping);
 
+            lctx.create_generic_defs(opaque_ty_def_id, &ast_generics.params, &mut new_remapping);
+            debug!(?new_remapping);
+
+            debug!("impl_trait_inputs = {:#?}", impl_trait_inputs);
+            for ty in impl_trait_inputs {
+                if let TyKind::ImplTrait(node_id, _) = &ty.kind {
```

Comment:
Why not just store the `NodeId`s and bounds in `impl_trait_inputs`?

Reply:
Yes, I had a fixme to do that at some point. Going to make this change.